yash-builtin/src/kill/syntax.rs

```diff
@@ -0,0 +1,849 @@
+    let mut signal_origin = None;
+    let mut list = None;
+    let mut verbose = None;
+
+    // Parse options
+    while let Some(arg) =
+        args.next_if(|arg| arg.value.strip_prefix('-').is_some_and(|s| !s.is_empty()))
+    {
+        let options = &arg.value[1..];
+        if options == "-" {
+            debug_assert_eq!(arg.value, "--");
+            break;
+        }
+
+        let mut chars = options.chars();
+        while let Some(option) = chars.next() {
+            match option {
+                's' | 'n' => {
+                    let remainder = chars.as_str();
+                    if remainder.is_empty() {
+                        let Some(current_signal_arg) = args.next() else {
+                            return Err(Error::MissingSignal {
+                                signal_option_name: option,
+                                signal_option_location: arg.origin,
+                            });
+                        };
+                        set_signal(
+                            &mut signal,
+                            &mut signal_origin,
+                            parse_signal_name_or_number(
+                                &current_signal_arg.value,
+                                allow_sig_prefix,
+                            ),
+                            current_signal_arg,
+                        )?;
+                    } else {
+                        set_signal(
+                            &mut signal,
+                            &mut signal_origin,
+                            parse_signal_name_or_number(remainder, allow_sig_prefix)
+                                .or_else(|| parse_signal_name_or_number(options, allow_sig_prefix)),
+                            arg,
+                        )?;
+                    }
+                    break;
+                }
+                'l' => {
+                    list = Some(arg.origin.clone());
+                }
+                'v' => {
+                    verbose = Some(arg.origin.clone());
+                }
+                _ => {
+                    set_signal(
+                        &mut signal,
+                        &mut signal_origin,
+                        parse_signal_name_or_number(options, allow_sig_prefix),
+                        arg,
+                    )
+                    .map_err(invalid_signal_to_unknown_option)?;
+                    break;
+                }
+            }
+        }
+    }
+
+    // Parse operands and compute the result
+    if let Some(option_location) = verbose {
+        parse_list_case(
+            args,
+            allow_sig_prefix,
+            signal_origin,
+            'v',
+            option_location,
+            true,
+        )
+    } else if let Some(option_location) = list {
+        parse_list_case(
+            args,
+            allow_sig_prefix,
+            signal_origin,
+            'l',
+            option_location,
+            false,
+        )
+    } else {
```

Comment:
Actually, no clippy warnings are issued for this part because of the comment between the statements. I think the current code structure annotated with the comment is more helpful to understand the intent of the code than collapsing the code block.